logger/gorm_logger.go

```diff
@@ -6,36 +6,31 @@ import (
 	"time"
 
 	"github.com/sirupsen/logrus"
+	"gorm.io/gorm"
 	gormLogger "gorm.io/gorm/logger"
 	"gorm.io/gorm/utils"
 )
 
-type CustomGORMLogger struct {
+type GormLogger struct {
 	Logger                  *logrus.Logger
 	SlowThreshold           time.Duration
 	SkipErrorRecordNotFound bool
-	LogLevelForSqlLogs      string
 }
 
-func (l *CustomGORMLogger) LogMode(gormLogger.LogLevel) gormLogger.Interface {
-	return l
-}
-
-func (l *CustomGORMLogger) Info(_ context.Context, logMessage string, data ...interface{}) {
-	l.Logger.Info(logMessage, data)
-}
+func (l *GormLogger) LogMode(gormLogger.LogLevel) gormLogger.Interface { return l }
 
-func (l *CustomGORMLogger) Warn(_ context.Context, logMessage string, data ...interface{}) {
-	l.Logger.Warn(logMessage, data)
-}
-
-func (l *CustomGORMLogger) Error(_ context.Context, logMessage string, data ...interface{}) {
-	l.Logger.Error(logMessage, data)
-}
-
-func (l *CustomGORMLogger) Trace(_ context.Context, begin time.Time, fc func() (sql string, rowsAffected int64), err error) {
-	var ErrorRecordNotFound = errors.New("record not found")
+// Functions implementing the rest of the gorm logger interface - not used
+// unless you were to call `DB.Logger.Info(...)`, which we don't use. Everything
+// else goes through Trace
+func (l *GormLogger) Debug(_ context.Context, logMessage string, data ...interface{}) {}
+func (l *GormLogger) Info(_ context.Context, logMessage string, data ...interface{})  {}
+func (l *GormLogger) Warn(_ context.Context, logMessage string, data ...interface{})  {}
+func (l *GormLogger) Error(_ context.Context, logMessage string, data ...interface{}) {}
 
+// Trace runs a SQL query and logs how long it took as well as the sql executed.
+// By default the log entry is debug, but if the SQL is very slow it will log as
+// warn.
+func (l *GormLogger) Trace(_ context.Context, begin time.Time, fc func() (sql string, rowsAffected int64), err error) {
```

Comment:
Regarding this one: so, does Gorm always call trace when logging SQL queries? On the comment above you say that in order to use any other level you must manually call `Debug|Info|Warn|Error`, so I am assuming that the answer is "yes", but...